libs/partners/ai21/langchain_ai21/contextual_answers.py

```diff
@@ -0,0 +1,76 @@
+from typing import (
+    Any,
+    Optional,
+    Type,
+    TypedDict,
+)
+
+from langchain_core.documents import Document
+from langchain_core.runnables import RunnableConfig, RunnableSerializable
+from typing_extensions import TypeAlias
+
+from langchain_ai21.ai21_base import AI21Base
+
+ANSWER_NOT_IN_CONTEXT_RESPONSE = "Answer not in context"
+
+
+class ContextualAnswerInput(TypedDict):
+    context: str
+    question: str
+
+
+class AI21ContextualAnswers(RunnableSerializable[ContextualAnswerInput, str], AI21Base):
+    class Config:
+        """Configuration for this pydantic object."""
+
+        arbitrary_types_allowed = True
+
+    @property
+    def InputType(self) -> TypeAlias:
+        """Get the input type for this runnable."""
+        return ContextualAnswerInput
+
+    @property
+    def OutputType(self) -> Type[str]:
+        """Get the input type for this runnable."""
+        return str
+
+    def _convert_input(self, input: ContextualAnswerInput) -> ContextualAnswerInput:
+        if input.get("context") is None or input.get("question") is None:
+            raise ValueError(
+                f"Input must contain a 'context' and 'question' field. Got {input}"
+            )
+
+        context_input = input["context"]
+
+        if isinstance(context_input, list) and context_input:
+            docs = [
+                item.page_content if isinstance(item, Document) else item
+                for item in context_input
+            ]
+            return {"context": "\n".join(docs), "question": input["question"]}
+
+        if isinstance(context_input, str):
+            return input
```

Comment:
This will check will fail if "question"'s type is int

Reply:
I don't think I need to validate the entire scale of input types the user may or may not pass as we are using a `TypedDict`. 
If the user decides to pass `{"question": 12, "context": "some context" }` the IDE will show them a warning and they will later receive an error/null from the server if they decide to ignore the initial warning.

The reason I'm checking the `input` type is because it could be two different things and I want to handle them differently